R/g.loadlog.R

```diff
@@ -7,7 +7,7 @@ g.loadlog = function(loglocation = c(), coln1 = c(), colid = c(),
   # Load sleep log data...
   S = data.table::fread(loglocation, stringsAsFactors = FALSE, data.table = FALSE,
                         check.names = TRUE, colClasses = "character")
-  nnights = floor((ncol(S) - coln1 + 1)) / 2
+  nnights = floor((ncol(S) - coln1 + 1) / 2)
```

Comment:
How about we change this to:

```
nnights = (ncol(S) - coln1 + 1) / 2
if (nnights%%2 == 1) { # test whether this an odd number
  warning(paste0("\nWe see an odd number of timestamp columns",
  " in the sleeplog. The last column will be ignored. If this is incorrect,",
  " please check that argument coln1 is correctly specified if you use a basic sleeplog format and",
  " that all days have a date column if you use an advanced sleeplog format.")
  nnights = floor(nnights)
}
```

Reply:
Sounds good, but nnights for advanced sleeplog is redefined in line 49, so I would include this after the code for the advanced sleeplog, i.e., in line 203

Reply:
`nnights` is used a few lines earlier to decide on the dimension of a matrix, so the correction needs to happen before that, right?

Reply:
Right, I have just moved it

Reply:
Sorry to keep being picky, but better to get it right in one go then to have to revisit this later.

Can the line `nnights = min(floor((ncol(S) - coln1 + 1) / 2), nnights)` on line 217 be  removed?

By that time `nnights` is already defined as either `floor((ncol(S) - coln1 + 1) / 2)` when it was a basic sleeplog or as the number of dates in the advanced sleeplog.

Reply:
Yes, I had the same wonder, but I didn't remove that line because I was not sure if that related to a specific scenario/study. But I can't see where this would be relevant as nnights is already defined at that point. I will remove it now.